src/app.js

```diff
@@ -6,8 +6,189 @@ import _ from 'underscore';
+const reservationFields = ['name', 'email', 'age']
+
+const render = function render(event) {
+  const tableElement = $('#trip-list');
+  tableElement.html('');
+
+  tripList.forEach((trip) => {
+    if (!trip.get('id')) {
+      console.log('this is a bad trip');
+      console.log(trip);
+    } else {
+      const generatedHTML = tripTemplate(trip.attributes);
+      tableElement.append(generatedHTML);
+    }
+  });
+
+  $('th.sort').removeClass('current-sort-field');
+  $(`th.sort.${ tripList.comparator }`).addClass('current-sort-field');
+
+};
+
+const showTrips = function showTrips(event) {
+  $('#trips-table').show();
+  $('.centerbutton').hide();
+};
+
+const showTripDetails = function showTripDetails(event) {
+  const trip = tripList.get($(this).attr('data-id'))
+
+  trip.fetch({
+    success: (model, response) =>{
+      // console.log('trip fetch success');
+      const generatedHTML = tripDetails(trip.attributes);
+      $('#trip-details').html(generatedHTML)
+    },
+    failure: (model, response) => {
+      console.log('trip fetch failure');
+      reportStatus('success', 'Trip temporarilty unavailable');
+    }
+  });
+};
+
+const loadRegistrationForm = function loadRegistrationForm(event) {
+  const tripId = {tripId: $(this).attr('data-id')}
+  const generatedHTML = registrationTemplate(tripId);
+  // i keep passing this id through into various html attributes - is this the best way of accessing it?
+  $('.sign-up').hide();
+  $('#trip-details').append(generatedHTML);
+};
+
+const loadModal = function loadModal(event) {
+  $('#add-trip').show();
+};
+
+const makeObject = function makeObject(fields, formId) {
+  const objectData = {};
+  fields.forEach((field) => {
+    const inputElement = $(`${formId} input[name="${ field }"]`);
+    const value = inputElement.val();
+    objectData[field] = value;
+
+    inputElement.val('');
+  });
+  return objectData
+};
+
+const addReservationHandler = function addReservationHandler(event) {
+  event.preventDefault();
+
+  const reservation = new Reservation(makeObject(reservationFields, '#res-form'));
+  reservation.tripId =$(this).attr('data-id')
+
+  // if (!reservation.isValid()) {
+  //   handleValidationFailures(reservation.validationError);
+  //   return;
+  // }
+
+  reservation.save({}, {
+    success: (model, response) => {
+      reportStatus('success', 'Successfully reserved spot on trip')
+    },
+    error: (model, response) => {
+      handleValidationFailures(response.responseJSON['errors']);
+    }
+  });
+};
+
+const addTripHandler = function addTripHandler(event) {
+  event.preventDefault();
+
+  const trip = new Trip(makeObject(tripFields, '#add-trip-form'));
+
+  if (!trip.isValid()) {
+    handleValidationFailures(trip.validationError);
+    return;
+  }
+
+  tripList.add(trip);
+
+  trip.save({}, {
+    success: (model, response) => {
+      reportStatus('success', 'Successfully saved book!');
```

Comment:
I would suggest at this point refreshing the `tripList`  with `fetch`, that would trigger an `update` event causing it to rerender with the new trip!